client.go

```diff
@@ -23,9 +35,43 @@ func (c client) doRequest(ctx context.Context, in *request, out interface{}) err
 	if err != nil {
 		return err
 	}
-	bytes, err := c.get(ctx, url)
+
+	resp, err := c.get(ctx, url)
 	if err != nil {
 		return err
 	}
-	return json.Unmarshal(bytes, out)
+	defer closeSilently(resp.Body)
+
+	// OSRM returns both codes 200 and 400 in a case with a body.
+	// In other cases, it returns an unexpected error without a body.
+	// http://project-osrm.org/docs/v5.5.1/api/#requests
```

Comment:
Responses link seems more relevant in this context - http://project-osrm.org/docs/v5.5.1/api/#responses

Reply:
agree, updated